lib/message/encrypt.js

```diff
@@ -15,5 +15,24 @@ export default function encryptMessage(options) {
     options.date = typeof options.date === 'undefined' ? serverTime() : options.date;
     options.compression = options.compression ? openpgp.enums.compression.zlib : undefined;
 
-    return openpgp.encrypt(options);
+    if (options.detached) {
+        // Create detached signature of message
+        const signature = await options.message.signDetached(options.privateKeys);
+
+        // Encrypt message without signing it (thus no need to change .detached)
+        options.privateKeys = [];
+        const result = await openpgp.encrypt(options);
+
+        // Encrypt signature and add it to the final result
+        options.message = createMessage(signature.packets.write(), options.filename);
+        const encryptedSignature = await openpgp.encrypt(options);
+        result.encryptedSignature = options.armor ? encryptedSignature.data : encryptedSignature.message;
```

Comment:
Oh, and finally, this doesn't quite work (your test currently fails) because `options.armor` is undefined by default in this function, whereas it's true by default inside `openpgp.encrypt`. So you have to change this function's signature to something like `function encryptMessage({ armor = true, ...options })`, and then pass `armor` back into the calls to `openpgp.encrypt`.